src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,44 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo | null;
+  setModalOpen: (val: boolean) => void;
+  resetSelectedTodo: (val: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  setModalOpen,
+  todo,
+  resetSelectedTodo,
+}) => {
+  const [fetchedUser, setFetchedUser] = useState<User | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (todo) {
+      getUser(todo.userId)
+        .then(setFetchedUser)
+        .catch(err => {
+          setError(`Error fetching user: ${err.message}`);
+          setFetchedUser(null);
+        });
+    }
+  }, [todo]);
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+    resetSelectedTodo(null);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {fetchedUser === null ? (
         <Loader />
```

Comment:
The `Loader` component is displayed when `fetchedUser` is null, which is appropriate for indicating a loading state. Ensure that the `Loader` is styled and positioned correctly to provide a good user experience.